src/layouts/MainLayout.vue

```diff
@@ -349,30 +357,45 @@ export default {
           console.warn('Non-Ethereum browser detected. You should consider trying MetaMask!')
           return
         }
-      } else if (wallet === 'phantom') {
-        const isPhantomInstalled = window.phantom?.solana?.isPhantom
-        if (isPhantomInstalled) {
-          const getProvider = () => {
-            if ('phantom' in window) {
-              provider = window.phantom?.solana
-              if (provider?.isPhantom) {
-                return provider
-              }
+      } else if (wallet === 'phantom' || wallet === 'solflare') {
+        const providers = {
+          phantom: () => {
+            return {
+              url: 'https://phantom.app/',
+              provider: wallet === 'phantom' ? window.phantom?.solana : null,
+              installed: window.phantom?.solana?.isPhantom
+            }
+          },
+          solflare: () => {
+            return {
+              url: 'https://solflare.com',
+              provider: wallet === 'solflare' ? window.solflare : null,
+              installed: window.solflare?.isSolflare
             }
-            window.open('https://phantom.app/', '_blank')
           }
```

Comment:
Why are the `providers` object values functions that return an object, rather than plain objects? Also why the double check on L365 and L372 (since the condition is already handled on L377) ?

Reply:
~~Why are the providers object values functions that return an object~~
L365 and L372 are not the same provider

Reply:
What I meant is that, since `providerInfo` is created from instantiating an object using the `wallet` variable, you don't need to add that condition inside of nested objects, using your code we'll have:

```js
const providerInfo = providers[wallet]
// if wallet === 'phantom'
// then :
providerInfo = {
  url: 'https://phantom.app/',
  provider: wallet === 'phantom' ? window.phantom?.solana : null,
  installed: window.phantom?.solana?.isPhantom
}

// if wallet === 'solflare'
// then :
providerInfo = {
  url: 'https://solflare.com',
  provider: wallet === 'solflare' ? window.solflare : null,
  installed: window.solflare?.isSolflare
}
```